website_product_document_download_counter/models/product_document.py

```diff
@@ -0,0 +1,44 @@
+# Copyright (C) 2024 Cetmix OÜ
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import logging
+
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+logger = logging.getLogger(__name__)
+
+
+class ProductDocument(models.Model):
+    _inherit = "product.document"
+
+    download_count = fields.Integer(
+        default=0,
+        help="Number of times this document has been downloaded",
+    )
+
+    count_downloads = fields.Boolean(
+        default=False,
+        help="Enable download counting for this document when published on website",
+    )
+
+    def increment_download_count(self):
+        """Increment the download count for a specific product document."""
+        # First, validate with the current environment
+        logger.info("Incrementing download count for %s", self.id)
```

Comment:
This message doesn't contain any valuable information, nothing special happens when a log counter is incremented, so we don't need it.